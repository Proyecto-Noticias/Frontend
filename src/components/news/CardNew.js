import React from "react";
import { Link } from "react-router-dom";
import noImage from "../../assets/notphoto.jpg";
//Add an aria-role at Link
// focus - style;
const CardNew = ({ imageUrl, title, _id, sentiment }) => {
  return (
    <Link className="focus__article" title={title} to={`/new/${_id}`}>
      <article className="card__container">
        <img
          src={imageUrl || noImage}
          alt="Portada de la noticia"
          loading="lazy"
          className="card__container--img"
        />
        <img
          className="card__container--sentiment"
          src={`../../assets/${sentiment}.png`}
          alt="Sensitive calification"
          loading="lazy"
        />
        <h2 className="card__items--label">{title}</h2>
      </article>
    </Link>
  );
};

export default CardNew;
