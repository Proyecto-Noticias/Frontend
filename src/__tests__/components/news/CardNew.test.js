import React from 'react';
import { shallow } from "enzyme";
import CardNew from '../../../components/news/CardNew';


describe('Test of <CardNew />', () => {
  const imageUrl = 'https://localhost/imgtest.jpg';
  const title = ' Test Title ';
  const _id = '123abc';
  
  const wrapper = shallow( <CardNew 
    imageUrl={imageUrl}
    title={title}
    _id={_id}
    /> )  
  
  test('should show the component correctly', () => {   
    expect ( wrapper ).toMatchSnapshot()
  })
  
  test('should show a title', () => {   

    const h2 = wrapper.find('h2')
    expect ( h2.text()).toBe(title)
  }) 

  test('should show a title', () => {   

    const h2 = wrapper.find('h2')
    expect ( h2.text()).toBe(title)
  }) 
  
  it("renders an image", () => {  
    expect(wrapper.find("img").at(0).prop("src")).toEqual(imageUrl)
  })

  it("redirects to news/:id", () => {  
    expect(wrapper.find("Link").prop("to")).toEqual(`/new/${_id}`)
  })
})