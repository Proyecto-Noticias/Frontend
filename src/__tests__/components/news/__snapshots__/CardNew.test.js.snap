// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <CardNew /> should show the component correctly 1`] = `
<Link
  className="focus__article"
  title=" Test Title "
  to="/new/123abc"
>
  <article
    className="card__container"
  >
    <img
      alt="Portada de la noticia"
      className="card__container--img"
      loading="lazy"
      src="https://localhost/imgtest.jpg"
    />
    <img
      alt="Sensitive calification"
      className="card__container--sentiment"
      loading="lazy"
      src="../../assets/undefined.png"
    />
    <h2
      className="card__items--label"
    >
       Test Title 
    </h2>
  </article>
</Link>
`;
